pkgs/development/python-modules/smpp.pdu/default.nix

```diff
@@ -0,0 +1,39 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, poetry-core
+, pydantic
+, requests
+, pytestCheckHook
+, pytest-asyncio
+, setuptools
+}:
+
+buildPythonPackage {
+  pname = "smpp.pdu";
+  version = "unstable-2022-09-02";
+  format = "pyproject";
+
+  # Upstream was once mozes/smpp.pdu, but it's dead and Python 2 only.
+  src = fetchFromGitHub {
+    owner = "hologram-io";
+    repo = "smpp.pdu";
+    rev = "20acc840ded958898eeb35ae9a18df9b29bdaaac";
+    hash = "sha256-/icVexc2S8sbJqn4ioeIhYxyDFIENuCfsFhl0uAHa9g=";
+  };
+
+  nativeBuildInputs = [
+    setuptools
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+  ];
+
```

Comment:
you probably want to add `pythonImportsCheck = [ "smpp.pdu" ];` here or what ever the import is called.

Reply:
Done.